### DISCUSSION TITLE
Focus navigation on Android TV

### DISCUSSION THREAD

**Discussion by vkotov** [AvaloniaUI/Avalonia#13326] [Q&A] [answered]
How to enable focus navigation on Android TV using directional pad?
Is this functionality already implemented or I need use KeyDown event and implement focus change by myself?

UPD: I tried using KeyDown event to move focus with `KeyboardNavigationHandler.Move`  but it works only when some element in view focused manually and after second key press. Also when I press SET key on directional pad then Avalonia provide Key.None value in event.
![image](https://github.com/AvaloniaUI/Avalonia/assets/1422494/ce6f657e-ae87-4159-9ce7-33cfbe019c0a)

**Comment by robloo** [comment, marked as answer]
So this really requires gamepad-type support to work out-of-the box. (Plus whatever Android-specific input drivers are needed). Gamepad navigation isn't completed yet.

 * https://github.com/AvaloniaUI/Avalonia/issues/6945
 * https://github.com/AvaloniaUI/Avalonia/issues/7607

> Is this functionality already implemented or I need use KeyDown event and implement focus change by myself?

Yes, I think you need to do this yourself... at a lower level than KeyDown unfortunately

**Reply by cyraid** [reply]
@robloo Why would you need Gamepad when Keycode.DpadLeft is a part of the OnKeyDown in Android (TV)?

**Reply by maxkatz6** [reply]
@cyraid focus navigation was implemented for android previously: https://github.com/AvaloniaUI/Avalonia/pull/13947
Including android key mapping to avalonia logical key enum (Keycode.DpadLeft -> Key.Left).

**Reply by cyraid** [reply]
@maxkatz6 really? So what you mean is I should ditch MAUI and say welcome back to Avalonia?
